lib/cdo/aws/s3.rb

```diff
@@ -314,29 +314,23 @@ def initialize(bucket, prefix, make_public=false)
       # @param [Hash] options
       # @return [String] public URL of uploaded log
       # @raise [Exception] if the S3 upload fails
-      # @see http://docs.aws.amazon.com/sdkforruby/api/Aws/S3/Client.html#put_object-instance_method for supported options
+      # @see https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/S3/Client.html#put_object-instance_method for supported options
       def upload_log(name, body, options={})
         key = "#{@prefix}/#{name}"
 
+        options[:content_type] ||= 'text/plain'
         options[:acl] = 'public-read' if @make_public
         result = AWS::S3.create_client.put_object(
           options.merge(
             bucket: @bucket,
             key: key,
             body: body,
-            content_type: 'text/plain'
           )
         )
-        if @make_public
-          log_url = AWS::S3.public_url(@bucket, key)
-          log_url += "?versionId=#{result[:version_id]}" unless result[:version_id].nil?
-          log_url
-        else
-          # Expire link in 72 hours
-          options = {bucket: @bucket, key: key, expires_in: 259200}
-          options[:version_id] = result[:version_id] unless result[:version_id].nil?
-          Aws::S3::Presigner.new.presigned_url(:get_object, options)
-        end
+
+        log_url = AWS::S3.public_url(@bucket, key)
```

Comment:
I believe this will create an unusable URL unless the object is public. For example, this "main.json" file will show you an access denied, even if you're logged in and otherwise have access to view and download the object.

Object URL: https://cdo-v3-sources.s3.amazonaws.com/sources/10000004/32640464/main.json

Console URL: https://s3.console.aws.amazon.com/s3/object/cdo-v3-sources?region=us-east-1&prefix=sources/10000004/32640464/main.json